themes/src/main/resources/theme/base/login/resources/js/authChecker.js

```diff
@@ -1,5 +1,7 @@
 const CHECK_INTERVAL_MILLISECS = 2000;
+const AUTH_SESSION_TIMEOUT_MILLISECS = 1000;
 const initialSession = getSession();
+const initialKcAuthSessionHash = getKcAuthSessionHash();
```

Comment:
I wonder if there is some corner case scenario like this: 

1) Authentication request (request1) in tab1 called and sets the cookie `KC_AUTH_SESSION_HASH=id1`
2) Authentication request (request2) in tab2 called and sets the cookie `KC_AUTH_SESSION_HASH=id2`
3) Tab1 calls the `authChecker.js` javascript. But since the javascript is called little bit later than first request from step1, it already reads the cookie with the new value `id2` set by different tab `tab2`. So javascript in tab1 won't detect the change and hence the bug will still happen.

I am not 100% sure if this scenario can happen, but I think that yes as the response from request1 doesn't call the javascript `authChecker.js` and hence there is some small window between them?

Maybe can be prevented by rendering the hashed/encrypted authSessionId also in some hidden element on the web page, so the `authChecker.js` can read the `initialKcAuthSessionHash` from this element instead of from the cookie?

Reply:
you're right that this scenario while rare can happen. I had already experimented with this solution by adding the hash in the AuthenticationSessionBean, reading it on the web page, and comparing it with the cookie after the timeout.

The only thing I didn’t like was hashing the auth session id every time the web page is loaded because I wasn't sure how to store it, since a new auth session is created and the root auth session doesn't have any auth notes. Would be fine hashing the auth session id every time on the web page? Or should we find a way to temporarily store it?

Reply:
Good point... I guess that something like this:
- The request, which did a creation of new authenticationSession and which set the AUTH_SESSION_ID and HASH cookie can also save the hashed attribute somewhere? Maybe to `KeycloakSession` as an attribute? Then the bean can read it.
- The request, which just relies on the existing AUTH_SESSION_ID (and existing HASH cookie) can just read it from this cookie.

Could something like this work? If yes, we don't need to hash multiple times per request?

Reply:
From the tests I've run, the hash has very little impact, so it doesn't seem necessary to find a way to cache the hash value.

Reply:
Sounds good, Thanks